entities/src/main/java/com/flipkart/varadhi/entities/Endpoint.java

```diff
@@ -0,0 +1,32 @@
+package com.flipkart.varadhi.entities;
+
+import jakarta.ws.rs.HttpMethod;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import java.net.URL;
+
+public abstract sealed class Endpoint {
+
+    abstract Protocol getProtocol();
+
+    @EqualsAndHashCode(callSuper = true)
+    @Data
+    public static final class HttpEndpoint extends Endpoint {
+        private final URL url;
+        private final HttpMethod method;
+        private final String contentType;
```

Comment:
Is this for default value in case it is not available with message ? If yes, should it be with topic instead of subscription ?

Reply:
Hmm.. I actually didn't put much thought here :)
Topic defining the content-type makes complete sense.

Will remove in the future PR for now.